avro/src/schema.rs

```diff
@@ -2761,7 +2766,44 @@ mod tests {
 
         let schema_str_c = r#"["A", "B"]"#;
 
-        let schema_c = Schema::parse_str_with_list(schema_str_c, &[schema_str_a, schema_str_b])?;
+        let (schema_c, schemata) =
+            Schema::parse_str_with_list(schema_str_c, &[schema_str_a, schema_str_b])?;
+
+        let schema_a_expected = Schema::Record(RecordSchema {
```

Comment:
I think it would be a good idea to start using a builder for the non-primitive schema types and RecordFiled. There are many fields which have good defaults which could be omitted here. 
I'll file an issue!

Reply:
https://github.com/apache/avro-rs/pull/108